src/libraries/Microsoft.Bcl.Cryptography/tests/X509Certificates/X509CertificateKeyAccessorsTests.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography.SLHDsa.Tests;
+using System.Security.Cryptography.X509Certificates;
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    public static class X509CertificateKeyAccessorsTests
+    {
+        [ConditionalFact(typeof(SlhDsa), nameof(SlhDsa.IsSupported))]
+        public static void GetPublicKey()
+        {
+#if NET10_0_OR_GREATER
+            string certPem = PemEncoding.WriteString("CERTIFICATE", SlhDsaTestData.IetfSlhDsaSha2_128sCertificate);
+            using (X509Certificate2 cert = X509Certificate2.CreateFromPem(certPem))
```

Comment:
The test should also check that GetSlhDsaPublicKey is returning a public-only key when a private key is available.